plugins/VstEffect/VstEffect.cpp

```diff
@@ -79,32 +79,44 @@ VstEffect::VstEffect( Model * _parent,
 
 
 
-Effect::ProcessStatus VstEffect::processImpl(SampleFrame* buf, const fpp_t frames)
+auto VstEffect::processImpl() -> ProcessStatus
```

Comment:
Maybe, though that would require rewriting a lot more of the `RemotePlugin` class which owns and manages the shared memory, and I'm not sure what the benefit of that would be. It looks like Zyn can run both locally and with `RemotePlugin` which means it needs the option of using `RemotePlugin`'s shared memory as its buffer or providing its own buffer.

Allowing a plugin implementation to provide its own buffer interface (rather than using `AudioPluginBufferDefaultImpl` like most plugins) makes the design really flexible where it needs to be. For example, it would be beneficial to CLAP where a custom buffer implementation would probably be needed to implement its [in-place processing optimization](https://github.com/free-audio/clap/blob/main/include/clap/ext/audio-ports.h#L61:L64).

Reply:
OK, maybe we leave it like this for now. It's difficult to foresee (or even overview) the different buffer requirements. I'm fine with resolving this.